dataikuapi/dss/ml.py

```diff
@@ -643,6 +652,351 @@ def __repr__(self):
+        """
+        Gets the raw dictionary of the condition
+        :rtype: dict
+        """
+        return self._internal_dict
+
+    @property
+    def expected_class(self):
+        """
+        Returns the expected class or None if it is not defined. Assertion passes if the expected_valid_ratio
+        of rows predicted as expected_class is attained.
+        :rtype: str
+        """
+        if "expectedClass" in self._internal_dict:
+            return self._internal_dict["expectedClass"]
+        else:
+            return None
+
+    @expected_class.setter
+    def expected_class(self, expected_class):
+        self._internal_dict["expectedClass"] = expected_class
+
+    @property
+    def expected_valid_ratio(self):
+        """
+        Returns the ratio of valid rows to exceed for the assertion to pass. A row is considered valid if the prediction
+        is equal to the expected class for classification or in the expected range for regression.
+        :rtype: str
+        """
+        return self._internal_dict["expectedValidRatio"]
+
+    @expected_valid_ratio.setter
+    def expected_valid_ratio(self, expected_valid_ratio):
+        self._internal_dict["expectedValidRatio"] = expected_valid_ratio
+
+    @property
+    def expected_min(self):
+        """
+        Returns the min (included) of the expected range or None if it is not defined.
+        Assertion passes if the expected_valid_ratio of rows predicted between expected_min and expected_max (included) is attained.
+        :rtype: float
+        """
+        if "expectedMinValue" in self._internal_dict:
+            return self._internal_dict["expectedMinValue"]
+        else:
+            return None
+
+    @expected_min.setter
+    def expected_min(self, expected_min):
+        self._internal_dict["expectedMinValue"] = expected_min
+
+    @property
+    def expected_max(self):
+        """
+        Returns the max (included) of the expected range or None if it is not defined.
+        Assertion passes if the ratio of rows predicted between expected_min and expected_max (included) is attained.
+        :rtype: float
+        """
+        if "expectedMaxValue" in self._internal_dict:
+            return self._internal_dict["expectedMaxValue"]
+        else:
+            return None
+
+    @expected_max.setter
+    def expected_max(self, expected_max):
+        self._internal_dict["expectedMaxValue"] = expected_max
+
+
+class DSSMLAssertionsMetrics(object):
+    """
+    Object that represents the per assertion metrics for all assertions on a trained model
+    Do not create this object directly, use :meth:`DSSTrainedPredictionModelDetails.get_assertions_metrics()` instead
+    """
+    def __init__(self, data):
+        self._internal_dict = data
+
+    def get_raw(self):
+        """
+        Gets the raw dictionary of the assertions metrics
+        :rtype: dict
+        """
+        return self._internal_dict
+
+    def get_metric(self, assertion_name):
```

Comment:
Open question: `metric` or `metrics`? `metric` makes it easier / more obvious to distinguish per assertion vs all, but technically we return several metrics for each assertion…
We could also dodge the question and call it `get_result` / `DSSAssertionResult` 😄 (in which case its `result` property could be renamed `passes`).

Reply:
The question of `metric` vs `result` is impacting the variable name consistency in dip as well. IMO, we should decide which is better and use only one everywhere

Reply:
I have added my proposal in my latest review. For me it should be `DSSMLAssertionMetrics` & `get_assertion_metrics`, what do you think ?

Reply:
- Agree with `DSSMLAssertionMetrics` and the `_metrics` plural for a single assertion
- Arnaud initially named the method `get_assertion_metrics`, and I suggested `get_metrics` as to me it seemed redundant (`trained.get_assertions_metrics().get_assertion_metric("name")` → `trained.get_assertions_metrics().get_metric("name")`). However that opinion is loosely held, so if you both think the clarity makes it worth the redundancy, I'll yield.

Reply:
You convinced me with your example, let's go with `get_metrics` :-)